### PR TITLE
Any closed turf blocks aux base placement

### LINKED ISSUE #35276: Deconstructing any part of the aux base after deployment makes rock walls
Round ID: 82966

Reproduction: We blind-launched the base and then started expanding it for a few hours. Anytime we deconstructed a wall on the aux base an intact rock wall would replace it, and later an ash walker bombed the aux base with a fireball staff, and half the base was turned into rock walls.

### PR DESCRIPTION
**Pull request by ninjanomnom**
:cl: ninjanomnom
fix: You need to actually clear an area for the aux base, remove all walls (except rock walls).
/:cl:

fixes #35276

### CODE REVIEW

**Comment by DaedalusGame** [comment]
This is not a fix

**Comment by DaedalusGame** [comment]
This is a fix

File: code/modules/mining/aux_base.dm [hunk 1]
```diff
@@ -248,6 +248,21 @@ interface with the mining shuttle at the landing site if a mobile beacon is also
...
 
+/obj/docking_port/mobile/auxillary_base/takeoff(
+	list/old_turfs,
+	list/new_turfs,
+	list/moved_atoms,
+	rotation,
+	movement_direction,
+	old_dock,
+	area/underlying_old_area,
+	)
```
Comment (DaedalusGame) [inline code/modules/mining/aux_base.dm]: This looks really fucked up

Comment (ninjanomnom) [reply code/modules/mining/aux_base.dm]: it's fine

File: code/modules/mining/aux_base.dm [hunk 2]
```diff
@@ -161,7 +161,7 @@ interface with the mining shuttle at the landing site if a mobile beacon is also
 				return BAD_COORDS
 			if(!istype(place.loc, /area/lavaland/surface))
 				return BAD_AREA
-			if(disallowed_turf_types[place.type])
+			if(disallowed_turf_types[place.type] && !istype(place, /turf/closed/mineral))
```
Comment (Cyberboss) [inline code/modules/mining/aux_base.dm]: This should be done at the typecache level
```dm
var/static/list/disallowed_turf_types = typecacheof(list(
...
)) - typecacheof(/turf/close/mineral)
```

File: code/modules/mining/aux_base.dm (hunk 1 above)
Comment (Cyberboss) [reply code/modules/mining/aux_base.dm]: It's really not

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/mining/aux_base.dm (hunk 1 above)
Comment (duncathan) [reply code/modules/mining/aux_base.dm]: It's really not

**Comment by ninjanomnom** [comment]
Fixing this up now that I have some motivation

**Comment by Hatfish** [comment]
This will ruin the autodrop feature.

**Comment by DaedalusGame** [comment]
this PR should have its title and CL updated because that's not at all what it does

**Comment by ninjanomnom** [comment]
the changelog is correct

**Comment by DaedalusGame** [comment]
ok i guess

**Pull request merged**